### ISSUE TITLE
[iOS] ListView Header/HeaderTemplate changes DataContext incorrectly

### ISSUE THREAD

**Issue by Soap-141** [unoplatform/uno#13173]
### Current behavior

Using `ListView.HeaderTemplate` makes the application crash on iOS and `ListView.Header` is not working at all.

### Expected behavior

The `ListView.HeaderTemplate` and `ListView.Header` properties should work.

### How to reproduce it (as minimally and precisely as possible)

1. Download [ListViewHeader.zip](https://github.com/unoplatform/uno/files/12295209/ListViewHeader.zip)
2. Start the application on Windows or Android
3. Notice it works
4. Start the application on iOS
5. Notice it crashes, does nothing or has a black screen.

### Workaround

N/A

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

Latest stable.

### Affected platforms

iOS

### IDE

Visual Studio 2022

### IDE version

17.6.5

**Comment by Soap-141** [comment]
https://github.com/unoplatform/tradezero-private/issues/6

**Comment by Soap-141** [comment]
@jeromelaban I just tried using latest Uno 5 release and it's still an issue.

**Comment by Soap-141** [comment]
@takla21 can you link the backport that caused this into 4.9 please?

**Comment by jeromelaban** [comment]
@Soap-141 the issue you're facing for the sample you built is https://github.com/unoplatform/uno/issues/212, and it's caused by `this.DataContext = this`.

If you move all the DataContext in a separate class, the header is shown and the binding works with 4.9.45.

I'm closing this issue, but feel free to reopen if you have a specific behavior that is incorrect.

---
Here's the updated sample:
```csharp
public sealed partial class MainPage : Page
{
	Model _model = new();

        public MainPage()
	{
		this.InitializeComponent();
		this.DataContext = _model;

        }

	private void Button_Click(object sender, RoutedEventArgs e)
	{
		_ = Frame.Navigate(typeof(TestPage));
	}

	private void Button_Click_1(object sender, RoutedEventArgs e)
	{
		_model.ListViewHeader.SimpleBinding += 1;
	}
}

public class Model
{
        public List<int> Source { get; } = new List<int>();

        public ListViewHeader ListViewHeader { get; } = new ListViewHeader();
}
```

**Comment by Soap-141** [comment]
@jeromelaban Main issue that I'm trying to reproduce right now is that my Header DataContext is becoming null after navigation. Think it has been introduced by this https://github.com/unoplatform/uno/pull/12939/files because it wasn't happening before 4.9.45.

**Comment by Soap-141** [comment]
@jeromelaban Tagged you as asked in meeting.

**Comment by Soap-141** [comment]
@jeromelaban This issue is not really updated with what we talked behind the scenes and also not working for that client anymore but if they mentions it you know what it is.

**Comment by jeromelaban** [comment]
This issue is fixed in 4.10.34, 5.0 latest, available on nuget.
